### PR TITLE
Closes #5564: Fix typo on add-ons strings xml file

### LINKED ISSUE #5564: Fix typo on add-ons strings xml file
In many of our strings we have "addon" and should be "add-on"

### PR DESCRIPTION
**Pull request by Amejia481**
---

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by Amejia481** [comment]
cc: @flodolo

File: components/feature/addons/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -51,65 +51,65 @@
     <string name="mozac_feature_addons_permissions_top_sites_description">Access browsing history</string>
     <!-- Description for devtools permission. -->
     <string name="mozac_feature_addons_permissions_devtools_description">Extend developer tools to access your data in open tabs</string>
-    <!-- The version on an addon. -->
+    <!-- The version on an add-on. -->
```
Comment (flodolo) [inline components/feature/addons/src/main/res/values/strings.xml]: on -> of

File: components/feature/addons/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -51,65 +51,65 @@
...
+    <!-- This is displayed in a page where the user can see the installed and recommended(not installed yet) add-ons, this string indicates the not yet supported section. -->
     <string name="mozac_feature_addons_unsupported_section">Not yet supported</string>
-    <!-- Displays a page with all the details of an addon. -->
+    <!-- Displays a page with all the details of an add-on. -->
     <string name="mozac_feature_addons_details">Details</string>
-    <!-- Displays a page with all the permissions of an addon. -->
+    <!-- Displays a page with all the permissions of an add-on. -->
     <string name="mozac_feature_addons_permissions">Permissions</string>
-    <!-- This is button that remove (uninstall an addon). -->
+    <!-- This is button that remove (uninstall an add-on). -->
```
Comment (flodolo) [inline components/feature/addons/src/main/res/values/strings.xml]: The closing parenthesis should be after uninstall?

File: components/feature/addons/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -51,65 +51,65 @@
...
     <string name="mozac_feature_addons_remove">Remove</string>
-    <!-- This is the title of a dialog that indicates to the user if wants to install or not an addon. %1$s is the addon name. -->
+    <!-- This is the title of a dialog that indicates to the user if wants to install or not an add-on. %1$s is the add-on name. -->
```
Comment (flodolo) [inline components/feature/addons/src/main/res/values/strings.xml]: asks the user if they want

File: components/feature/addons/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -51,65 +51,65 @@
...
     <string name="mozac_feature_addons_permissions_dialog_title">Add %1$s?</string>
-    <!-- This is the subtitle of a dialog that indicate to the user if wants to install or not an addon, the subtitle list the permissions that this addon requires, the permissions will dynamically listed after the ":". -->
+    <!-- This is the subtitle of a dialog that indicate to the user if wants to install or not an add-on, the subtitle list the permissions that this add-on requires, the permissions will dynamically listed after the ":". -->
```
Comment (flodolo) [inline components/feature/addons/src/main/res/values/strings.xml]: asks the user if they want

File: components/feature/addons/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -51,65 +51,65 @@
...
     <string name="mozac_feature_addons_permissions_dialog_subtitle">It requires your permission to:</string>
-    <!-- This a button to add (installs) an addon . -->
+    <!-- This a button to add (installs) an add-on . -->
```
Comment (flodolo) [inline components/feature/addons/src/main/res/values/strings.xml]: installs -> install

File: components/feature/addons/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -51,65 +51,65 @@
...
     <string name="mozac_feature_addons_permissions_dialog_add">Add</string>
-    <!-- This a button to cancel the addon installation . -->
+    <!-- This a button to cancel the add-on installation . -->
     <string name="mozac_feature_addons_permissions_dialog_cancel">Cancel</string>
-    <!-- Accessibility content description to install addon button. -->
+    <!-- Accessibility content description to install add-on button. -->
     <string name="mozac_feature_addons_install_addon_content_description">Install Add-on</string>
-    <!-- Indicate how many user has rated an addon. %1$s will be replace with number of users -->
+    <!-- Indicate how many user has rated an add-on. %1$s will be replace with number of users -->
```
Comment (flodolo) [inline components/feature/addons/src/main/res/values/strings.xml]: replace -> replaced

File: components/feature/addons/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -51,65 +51,65 @@
...
     <string name="mozac_feature_addons_user_rating_count">Users: %1$s</string>
-    <!-- Accessibility content description for the amount of stars that addon has, where %1$.02f will be the amount of stars and / separator and 5 the maximum number of stars e.g (2/5, 4.5/5 or 5/5)  . -->
+    <!-- Accessibility content description for the amount of stars that add-on has, where %1$.02f will be the amount of stars and / separator and 5 the maximum number of stars e.g (2/5, 4.5/5 or 5/5)  . -->
     <string name="mozac_feature_addons_rating_content_description">%1$.02f/5</string>
     <!-- This is the title of page where all the add-ons are listed-->
     <string name="mozac_feature_addons_addons">Add-ons</string>
-    <!-- The label of the allow button, this will be shown to the user when an addons needs new permissions, with the button the user will indicate that wants to accept the new permissions and update the addon-->
+    <!-- The label of the allow button, this will be shown to the user when an add-ons needs new permissions, with the button the user will indicate that wants to accept the new permissions and update the add-on-->
```
Comment (flodolo) [inline components/feature/addons/src/main/res/values/strings.xml]: an add-on
that wants -> that they want

File: components/feature/addons/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -51,65 +51,65 @@
...
+    <!-- The label of the deny button on a notification, this will be shown to the user when an add-ons needs new permissions. Indicates the user denies the new permissions and prevent the add-on from be updated-->
     <string name="mozac_feature_addons_updater_notification_deny_button">Deny</string>
-    <!-- The tile of the notification, this will be shown to the user when an addons needs new permissions, to be updated. %1$s is the name of the addons-->
+    <!-- The tile of the notification, this will be shown to the user when an add-ons needs new permissions, to be updated. %1$s is the name of the add-ons-->
     <string name="mozac_feature_addons_updater_notification_title">%1$s has a new update</string>
-    <!-- The content of the notification, this will be shown when an addons needs new permissions, to be updated. %1$d is the amount of new permissions-->
+    <!-- The content of the notification, this will be shown when an add-ons needs new permissions, to be updated. %1$d is the amount of new permissions-->
```
Comment (flodolo) [inline components/feature/addons/src/main/res/values/strings.xml]: an add-on

File: components/feature/addons/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -51,65 +51,65 @@
...
+    <!-- The label of the allow button, this will be shown to the user when an add-ons needs new permissions, with the button the user will indicate that wants to accept the new permissions and update the add-on-->
     <string name="mozac_feature_addons_updater_notification_allow_button">Allow</string>
-    <!-- The label of the deny button on a notification, this will be shown to the user when an addons needs new permissions. Indicates the user denies the new permissions and prevent the addon from be updated-->
+    <!-- The label of the deny button on a notification, this will be shown to the user when an add-ons needs new permissions. Indicates the user denies the new permissions and prevent the add-on from be updated-->
     <string name="mozac_feature_addons_updater_notification_deny_button">Deny</string>
-    <!-- The tile of the notification, this will be shown to the user when an addons needs new permissions, to be updated. %1$s is the name of the addons-->
+    <!-- The tile of the notification, this will be shown to the user when an add-ons needs new permissions, to be updated. %1$s is the name of the add-ons-->
```
Comment (flodolo) [inline components/feature/addons/src/main/res/values/strings.xml]: an add-on

File: components/feature/addons/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -51,65 +51,65 @@
...
     <string name="mozac_feature_addons_rating_content_description">%1$.02f/5</string>
     <!-- This is the title of page where all the add-ons are listed-->
     <string name="mozac_feature_addons_addons">Add-ons</string>
-    <!-- The label of the allow button, this will be shown to the user when an addons needs new permissions, with the button the user will indicate that wants to accept the new permissions and update the addon-->
+    <!-- The label of the allow button, this will be shown to the user when an add-ons needs new permissions, with the button the user will indicate that wants to accept the new permissions and update the add-on-->
     <string name="mozac_feature_addons_updater_notification_allow_button">Allow</string>
-    <!-- The label of the deny button on a notification, this will be shown to the user when an addons needs new permissions. Indicates the user denies the new permissions and prevent the addon from be updated-->
+    <!-- The label of the deny button on a notification, this will be shown to the user when an add-ons needs new permissions. Indicates the user denies the new permissions and prevent the add-on from be updated-->
```
Comment (flodolo) [inline components/feature/addons/src/main/res/values/strings.xml]: an add-on

**Comment by Amejia481** [comment]
Thanks @flodolo for the feedback :)
I updated the PR with your suggestions.

File: components/feature/addons/src/main/res/values/strings.xml [hunk 11]
```diff
@@ -51,65 +51,65 @@
...
+    <!-- Displays a page with all the permissions of an add-on. -->
     <string name="mozac_feature_addons_permissions">Permissions</string>
-    <!-- This is button that remove (uninstall an addon). -->
+    <!-- This is button that remove (uninstall an add-on). -->
     <string name="mozac_feature_addons_remove">Remove</string>
-    <!-- This is the title of a dialog that indicates to the user if wants to install or not an addon. %1$s is the addon name. -->
+    <!-- This is the title of a dialog that asks the users if they to install or not an add-on. %1$s is the add-on name. -->
     <string name="mozac_feature_addons_permissions_dialog_title">Add %1$s?</string>
-    <!-- This is the subtitle of a dialog that indicate to the user if wants to install or not an addon, the subtitle list the permissions that this addon requires, the permissions will dynamically listed after the ":". -->
+    <!-- This is the subtitle of a dialog that asks the users if they wants to install or not an add-on, the subtitle list the permissions that this add-on requires, the permissions will dynamically listed after the ":". -->
```
Comment (Delphine) [inline components/feature/addons/src/main/res/values/strings.xml]: nits: 
* if they *want* to install
* the subtitle *lists"
* will dynamically *be* listed

File: components/feature/addons/src/main/res/values/strings.xml [hunk 12]
```diff
@@ -51,65 +51,65 @@
...
     <string name="mozac_feature_addons_permissions_dialog_subtitle">It requires your permission to:</string>
-    <!-- This a button to add (installs) an addon . -->
+    <!-- This a button to add (install) an add-on . -->
```
Comment (Delphine) [inline components/feature/addons/src/main/res/values/strings.xml]: nit: this *is* a button

File: components/feature/addons/src/main/res/values/strings.xml [hunk 13]
```diff
@@ -51,65 +51,65 @@
...
     <string name="mozac_feature_addons_permissions_dialog_add">Add</string>
-    <!-- This a button to cancel the addon installation . -->
+    <!-- This a button to cancel the add-on installation . -->
```
Comment (Delphine) [inline components/feature/addons/src/main/res/values/strings.xml]: nit: this *is* a button

File: components/feature/addons/src/main/res/values/strings.xml [hunk 14]
```diff
@@ -51,65 +51,65 @@
...
     <string name="mozac_feature_addons_permissions_dialog_cancel">Cancel</string>
-    <!-- Accessibility content description to install addon button. -->
+    <!-- Accessibility content description to install add-on button. -->
     <string name="mozac_feature_addons_install_addon_content_description">Install Add-on</string>
-    <!-- Indicate how many user has rated an addon. %1$s will be replace with number of users -->
+    <!-- Indicate how many user has rated an add-on. %1$s will be replaced with number of users -->
```
Comment (Delphine) [inline components/feature/addons/src/main/res/values/strings.xml]: nits: 
* Indicate*s* how many user*s* *have*

File: components/feature/addons/src/main/res/values/strings.xml [hunk 15]
```diff
@@ -51,65 +51,65 @@
...
     <string name="mozac_feature_addons_rating_content_description">%1$.02f/5</string>
     <!-- This is the title of page where all the add-ons are listed-->
     <string name="mozac_feature_addons_addons">Add-ons</string>
-    <!-- The label of the allow button, this will be shown to the user when an addons needs new permissions, with the button the user will indicate that wants to accept the new permissions and update the addon-->
+    <!-- The label of the allow button, this will be shown to the user when an add-on needs new permissions, with the button the user will indicate that they want to accept the new permissions and update the add-on-->
     <string name="mozac_feature_addons_updater_notification_allow_button">Allow</string>
-    <!-- The label of the deny button on a notification, this will be shown to the user when an addons needs new permissions. Indicates the user denies the new permissions and prevent the addon from be updated-->
+    <!-- The label of the deny button on a notification, this will be shown to the user when an add-on needs new permissions. Indicates the user denies the new permissions and prevent the add-on from be updated-->
```
Comment (Delphine) [inline components/feature/addons/src/main/res/values/strings.xml]: Nit: and prevent*s*

**Comment by Amejia481** [comment]
Thanks @Delphine I updated the PR with your suggestions :)

**Comment by Amejia481** [comment]
Thank you so much for the detailed feedback :)
bors r=flodolo,delphine

**Comment by Amejia481** [comment]
bors retry

**Pull request merged**